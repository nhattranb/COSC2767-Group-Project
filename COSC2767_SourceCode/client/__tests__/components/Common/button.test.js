import React from 'react';
import { render } from '@testing-library/react';
import Button from '../../../app/components/Common/Button';

describe('Button component', () => {
    it('renders correctly with default props', () => {
        const { asFragment } = render(<Button>Default Button</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with primary variant', () => {
        const { asFragment } = render(<Button variant="primary">Primary Button</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with danger variant', () => {
        const { asFragment } = render(<Button variant="danger">Danger Button</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with custom className', () => {
        const { asFragment } = render(<Button className="custom-class">Custom Class Button</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with borderless and round props', () => {
        const { asFragment } = render(<Button borderless round={10}>Borderless Round Button</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with icon and text', () => {
        const { asFragment } = render(<Button icon={<span>Icon</span>} text="Button with Icon" />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with tooltip', () => {
        const { asFragment } = render(<Button tooltip tooltipContent="Tooltip Content">Button with Tooltip</Button>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly with popover', () => {
        const { asFragment } = render(<Button popover popoverContent="Popover Content" popoverTitle="Popover Title">Button with Popover</Button>);
        expect(asFragment()).toMatchSnapshot();
    });
});
