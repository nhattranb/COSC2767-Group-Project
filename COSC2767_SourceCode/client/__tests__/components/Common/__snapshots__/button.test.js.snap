// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly with borderless and round props 1`] = `
<DocumentFragment>
  <button
    class="input-btn custom-btn-secondary md text-only icon-left border-0"
    style="border-radius: 10px;"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button component renders correctly with custom className 1`] = `
<DocumentFragment>
  <button
    class="input-btn custom-class custom-btn-secondary md text-only icon-left "
    style="border-radius: 3px;"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button component renders correctly with danger variant 1`] = `
<DocumentFragment>
  <button
    class="input-btn custom-btn-danger md text-only icon-left "
    style="border-radius: 3px;"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button component renders correctly with default props 1`] = `
<DocumentFragment>
  <button
    class="input-btn custom-btn-secondary md text-only icon-left "
    style="border-radius: 3px;"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button component renders correctly with icon and text 1`] = `
<DocumentFragment>
  <button
    class="input-btn custom-btn-secondary md with-icon icon-left "
    style="border-radius: 3px;"
    type="button"
  >
    <div
      class="btn-icon"
    >
      <span>
        Icon
      </span>
    </div>
    <span
      class="btn-text"
    >
      Button with Icon
    </span>
  </button>
</DocumentFragment>
`;

exports[`Button component renders correctly with popover 1`] = `
<DocumentFragment>
  <button
    class="input-btn custom-btn-secondary md text-only icon-left "
    id="popover-undefined"
    style="border-radius: 3px;"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button component renders correctly with primary variant 1`] = `
<DocumentFragment>
  <button
    class="input-btn custom-btn-primary md text-only icon-left "
    style="border-radius: 3px;"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button component renders correctly with tooltip 1`] = `
<DocumentFragment>
  <button
    class="input-btn custom-btn-secondary md text-only icon-left "
    id="tooltip-undefined"
    style="border-radius: 3px;"
    type="button"
  />
</DocumentFragment>
`;
